### ISSUE TITLE
Crash when switching monitors via the mouse

### ISSUE THREAD

**Issue by t-wissmann** [herbstluftwm/herbstluftwm#482]
In the winterbreeze branch, sometimes I notice a segfault when switching monitors with `focus_follows_mouse`. I can't reproduce it yet.

**Comment by dnnr** [comment]
Are you running a Debug build with sanitizers enabled? I don't think I've ever seen a ASAN-enabled program actually segfault (it would mean that ASAN was fine with everything that led to there).

**Comment by t-wissmann** [comment]
I think it's sigabrt. So far I had two crashes, and dmesg told me:
```
$  dmesg -T|grep herbstl
[Do Mai  9 17:21:51 2019] audit: type=1701 audit(1557415280.614:11151): auid=1000 uid=1000 gid=100 ses=1 pid=906 comm="herbstluftwm" exe="/home/thorsten/git/herbstluftwm/debug/herbstluftwm" sig=6 res=1
[Do Mai  9 17:26:29 2019] audit: type=1701 audit(1557415558.335:11155): auid=1000 uid=1000 gid=100 ses=1 pid=7038 comm="herbstluftwm" exe="/home/thorsten/git/herbstluftwm/debug/herbstluftwm" sig=6 res=1
```
And I found the issue in my `~/.cache/lxsession/LXDE/run.log`:
```
herbstluftwm: /home/thorsten/git/herbstluftwm/src/layout.cpp:943: bool focus_client(Client*, bool, bool): Assertion `cur_mon == monitor' failed.
```

**Comment by t-wissmann** [comment]
It seems that I had an old version built. So I'm not even sure whether the issue still exists in current winterbreeze.

**Comment by t-wissmann** [comment]
Fixed by #484
